### ISSUE TITLE
 Add "Art Project" Option, and Add "Custom" Option, (NEW LEGO SET)

### ISSUE THREAD

**Issue by SilverHawk-1** [custombrickmosaic/custombrickmosaic.github.io#1]
Here is the code for the new option (to simplified the implementation!), sorry too lazy to fork!

case "ArtProject":
			var partList = [
				[108, 116, 211, 317, 0],
				[133, 139, 107, 127, 0],
                [204, 112, 42, 281, 0],
				[34,237,167,119,0],
				[228, 173, 200, 58, 0],
				[88, 42, 18, 307, 0],
				[248,187,61,175,0],
				[10,52,99,660,0],
				[187,233,11,166,0],
				[75,159,74,148,0],
				[254, 138, 24, 121, 0],
				[5, 19, 29, 254, 0],
				[200, 112, 160, 299, 0],
				[201, 26, 9, 190, 0],
				[228,205,158,241,0],
				[255, 255, 255, 474, 0]
				];
			return partList;
		case "Custom":
			var partList = [
				[0,143,155,99999,0],
				[5, 19, 29, 99999, 0],
				[10,52,99,99999,0],
				[34,237,167,99999,0],
				[53, 33, 0, 99999, 0],
				[54,174,191,99999,0],
				[75,159,74,99999,0],
				[88, 42, 18, 99999, 0],
                [96, 116, 161, 99999, 0],
				[108, 116, 211, 99999, 0],
				[108, 110, 104, 99999, 0],
				[114, 14, 15, 99999, 0],
				[133, 139, 107, 99999, 0],
                [149, 138, 115, 99999, 0],
				[159, 195, 233, 99999, 0],
				[160, 165, 169, 99999, 0],
				[169, 85,  0, 99999, 0],
				[187,233,11,99999,0],
				[200, 112, 160, 99999, 0],
				[201, 26, 9, 99999, 0],
               	[204, 112, 42, 99999, 0],
				[228, 173, 200, 99999, 0],
				[228, 205, 158, 99999, 0],
				[242, 205, 55, 99999, 0],
				[248,187,61,99999,0],
				[254, 138, 24, 99999, 0],
				[255,105,143,99999,0],
				[255, 255, 255, 99999, 0]
				];
			return partList;

and the HTML 👍 

<div class="form-group row mx-1 px-2">
						<div class="col-8">
							<label>ArtProject</label>
						</div>
						<div class="col-4">
							<div class="input-group input-group-sm mb-1">
								<button class="btn btn-secondary py-0" type="button" id="buttonMinusArtProject">-</button>
								<input type="text" class="form-control text-center py-0" id="inputArtProject" aria-label="numberOfArtProjectSet" placeholder="0" value="0">
								<button class="btn btn-secondary py-0" type="button" id="buttonPlusArtProject">+</button>
							</div>
						</div>
					</div>					
					<div class="form-group row mx-1 px-2">
						<div class="col-8">
							<label>Custom</label>
						</div>
						<div class="col-4">
							<div class="input-group input-group-sm mb-1">
								<button class="btn btn-secondary py-0" type="button" id="buttonMinusCustom">-</button>
								<input type="text" class="form-control text-center py-0" id="inputCustom" aria-label="numberOfCustomSet" placeholder="0" value="0">
								<button class="btn btn-secondary py-0" type="button" id="buttonPlusCustom">+</button>
							</div>
						</div>
					</div>

Also i want to say, your program is AWESOME!! thx a lot for it!!!

**Comment by SilverHawk-1** [comment]
sorry HTML did not work... anyway!, you understand!

**Comment by SilverHawk-1** [comment]
also, if i provide you the exact part number of each part (including the set) can you print those in the guide!? please! it will be really helpful!! thx!!

**Comment by custombrickmosaic** [comment]
Hi, thanks for your support + code suggestion. I added the "Art project". As for the "Custom" option, I hesitate to include it, as the type of the brick (plate or tile, round or square) also needs to be indicated and there would be several options per color, depending on the set. A suggestion would be to increase the number for all sets in step 3. Then you also have all colors available.

I do not fully understand your last comment. Do you mean the lego part number (e.g. 6284070 for the black round tile)?

**Comment by SilverHawk-1** [comment]
sorry it wasn’t clear!!, yes the lego part number and also the set number!, that way i would be able to know the exact part i need to order and also from wich set!!, because all color are not available to order unless they are order from the set! 
so it will be really awesome to include those on the first page of the instruction!


(for the custom set I on’y include the flat tile from all different set, (for me it was what i want), but i underssand it is not the reality for everyone else!!)


also, thx a lot!, your little program is amazing me and my girlfriend spend a lot of time trying to figure wich picture we will reproduce on our wall!!

 
 

Sébastien Ouellet
Intégrateur CAD / Chargé de projet
 
Gestion CAD | Programmation | Formation |Conception & Dessin | Technologie 4.0
514-402-1193 | ***@***.*** | www.mcmcad.com

**Comment by jestelle** [comment]
First of all @custombrickmosaic, AWESOME JOB on this project. I just opened the Art Project lego and have been messing in photoshop, downloading different apps, and not finding anything that did the job adequately (and easily) to take a custom image and make a nice lego mosaic. 

Your project finally got it right. Almost :D 

I found there's a couple bugs in the art project color list. This is my new list:

                                [5, 19, 29, 254, 0],
                                [10,52,99,660,0],
                                [42, 137, 255, 127, 0], 
                                [178, 241, 255, 317, 0], 
                                [255, 255, 255, 474, 0],
                                [228, 173, 200, 58, 0],
                                [201, 26, 9, 190, 0],
                                [254, 138, 24, 121, 0],
                                [248,187,61,175,0], 
                                [255, 235, 59, 299, 0], 
                                [253,245,172,119,0], 
                                [228,205,158,241,0],
                                [204, 112, 42, 281, 0],
                                [88, 42, 18, 307, 0],
                                [75,159,74,148,0],
                                [187,233,11,166,0],


Here's your original, notated with the problems:

				[108, 116, 211, 317, 0],  // This looks dark'ish blue, but should be a lighter blue.
				[133, 139, 107, 127, 0],  // This looks green'ish beige, but should be vibrant blue
				[204, 112, 42, 281, 0],
				[34,237,167,119,0], // This looks turquoise, but should be pale yellow
				[228, 173, 200, 58, 0],
				[88, 42, 18, 307, 0],
				[248,187,61,175,0],
				[10,52,99,660,0],
				[187,233,11,166,0],
				[75,159,74,148,0],
				[254, 138, 24, 121, 0],
				[5, 19, 29, 254, 0],
				[200, 112, 160, 299, 0], // This looks dark pink, but should be vibrant yellow
				[201, 26, 9, 190, 0],
				[228,205,158,241,0],
				[255, 255, 255, 474, 0]


Thanks again for the great project!

**Comment by custombrickmosaic** [comment]
Hi @jestelle. Thanks for your input! I checked all sets and colors again with the rebrickable database. Hence, the numbers I used are slightly different from your suggestions. Let me know if there is still something odd. Enjoy building the mosaics!

@SilverHawk-1. I added the lego part number to the pdf export. The set number is tricky, because some parts occur in more than one set. I hope it still helps.
